src/components/organisms/Top.tsx

```diff
@@ -2,24 +2,43 @@ import React, { useState, useEffect } from "react";
 import styles from "@/styles/components/organisms/Top.module.scss";
 import Weather from "@/components/molecules/Weather";
 import Livedata from "./Livedata";
+import axios from "axios";
 
 export default function Top() {
-  const [lat, setLat] = useState<number | null>();
-  const [long, setLong] = useState<number | null>();
   const [currTemp, setCurrTemp] = useState<number | null>();
   const [currHumid, setCurrHumid] = useState<number | null>();
+  const [deviceId, setDeviceId] = useState<string | null>();
+  const [weatherData, setWeatherData] = useState<any>([]);
 
```

Comment:
Change the type `any`.
Use the `WeatherData`.
But define the `WeatherData` in `Top.tsx` and export. And delete the `WeatherData` in `Weather.tsx` and import from `Top.tsx`.